### PR TITLE
Implement clean SVG solution

### LINKED ISSUE #2: retina display support for assets
## Problem

People use radically different resolution displays, and assets that look good on one don't look good on the other.
## Solution
- https://imulus.github.io/retinajs/
- resolution independent formats like svg

### PR DESCRIPTION
**Pull request by sandrina-p**
- Start by checking [Design System page - Illustrations](http://localhost:8000/app/design-system#Illustrations)
- To understand how it was implemented read `assets/svgs/README.md`.

If you don't care much about that part, just use them:
```
// pug
svg-hello

// js
import SvgHello from '@svgs/hello.svg'
```

- All SVGs go under `assets/svgs/original`
- Any question, feel free to ask!

Closes #2

### CODE REVIEW

**Comment by mmbotelho** [comment]
Awesome! Nice to see our illustrations organized 😄 

One question - how are you handling the attributions for the original creators of the illustrators?

**Review by pieer — DISMISSED** [review]

**Comment by sandrina-p** [comment]
@mmbotelho I have no idea... never saw it. Maybe some note in the settings page? Dunno... do you have any suggestions in mind?

**Comment by mmbotelho** [comment]
@sandrina-p here's a copy-paste of what I wrote on the design system (just above the illustrations):

> These are used on empty states / success screens / instructional sections. Most of the illustrations here are taken from thenounproject.com, and their authors need to be credited. Read more here: https://creativecommons.org/licenses/by/3.0/us/legalcode

> Attribution best practices:
https://wiki.creativecommons.org/wiki/best_practices_for_attribution

> In each one of these, I will write down what needs to be added to the description of the image (no need for it to be visible on the interface itself).

I don't know what is the best way to include that in our code, sorry!

**Comment by sandrina-p** [comment]
reading that doc, maybe we could:
- add it directly to the code next to the svg in-line
- add in to the terms page
- add it to the readme of the project

What do you think @taoeffect?

File: .Gruntfile.babel.js [hunk 1]
```diff
@@ -175,6 +226,8 @@ module.exports = (grunt) => {
   //  Grunt Tasks
   // -------------------------------------------------------------------------
 
+  grunt.loadNpmTasks('grunt-svg-sprite')
```
Comment (taoeffect) [inline .Gruntfile.babel.js]: I'm fairly certain this isn't necessary because of the `require('load-grunt-tasks')(grunt)` line. Can you double-check?

**Comment by taoeffect** [comment]
> What do you think @taoeffect?

For each illustration, make sure you meet the basic requirements of whatever license they use. If, for example, they state that attribution must be given, check if it also specifies *how* attribution must be given. Important for our purposes is the distinction between source and user-facing product/project.

At a minimum, you can list all of them in the README. But if further attribution is required (i.e. in the user-facing product/project), then yes, we'll need to add them to the app itself.

In fact, this might be worth doing anyway, regardless of whether or not the illustration requires it.

All apps have an "About" window. Ours should too.

I've created an issue for this: https://github.com/okTurtles/group-income-simple/issues/673

File: frontend/assets/svgs/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Illustrations as inline SVGs
+
+## What it is
```
Comment (taoeffect) [inline frontend/assets/svgs/README.md:3]: Missing newline

File: frontend/assets/svgs/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+### Optimization
+
+When running `grunt dev`, 2 things happen to SVGs: **compression** and **compiling**:
+
+1. **compression**
+
+A Grunt Task `svg_sprite` is run on build time only. In there, [SVGO](https://github.com/svg/svgo/), a tool to compress SVGs, comes to action. The final result is exported to `assets/svgs/compressed`. Do NOT modify any file there directly (note that it's even added to `.gitignore`).
+
+2. **compiling**
+A custom loader `svgLoader` is run during watch mode. This allow us to import SVGs directly and use them inline in the code. It also performs two additional things:
```
Comment (taoeffect) [inline frontend/assets/svgs/README.md]: Consistency here is needed. Either include a newline here or place the description on the same line for both.

I think creating a list is clearer personally.

```markdown
1. **compression** — A Grunt Task `svg_sprite` is run on build time only. In there, [SVGO](https://github.com/svg/svgo/), a tool to compress SVGs, comes to action. The final result is exported to `assets/svgs/compressed`. Do NOT modify any file there directly (note that it's even added to `.gitignore`).
2. **compiling** — A custom loader `svgLoader` is run during watch mode. This allow us to import SVGs directly and use them inline in the code. It also performs two additional things:
```

File: frontend/assets/svgs/README.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  - Remove `width` and `height` attributes from SVG
+  - Add a CSS class with the name of the icon `.svg-{file-name}`. This allow us to customize SVGs colors based on the current theme.
```
Comment (taoeffect) [inline frontend/assets/svgs/README.md]: Lists-within-lists should be indented

File: .Gruntfile.babel.js [hunk 5]
```diff
@@ -417,6 +473,38 @@ function flow (options = {}) {
   }
 }
 
+function svgLoader (options) {
+  // Based on "rollup-plugin-vue-inline-svg" but better:
+  // - fix sourcemaps bug on original plugin
+  // - remove useless attrs and add global CSS class.
```
Comment (taoeffect) [inline .Gruntfile.babel.js:477]: Very nice. You get several kudos here:

1. Excellent comments, they explain clearly where you got the inspiration for this from, and they explain your modifications/improvements
2. You avoided adding yet another unnecessary dependency to our codebase
3. Down below and up above you've linked to relevant Github issues as appropriate

Bravo! 👏

**Review by taoeffect — CHANGES_REQUESTED** [review]
Very clever PR @sandrina-p, well done, excellent work! 👏 

I've left comments on minor issues as I noticed them.

File: .Gruntfile.babel.js (hunk 1 above)
Comment (sandrina-p) [reply .Gruntfile.babel.js]: You are right, removed.

File: frontend/assets/svgs/README.md (hunk 2 above)
Comment (sandrina-p) [reply frontend/assets/svgs/README.md:3]: added.

File: frontend/assets/svgs/README.md (hunk 3 above)
Comment (sandrina-p) [reply frontend/assets/svgs/README.md]: done

File: frontend/assets/svgs/README.md (hunk 4 above)
Comment (sandrina-p) [reply frontend/assets/svgs/README.md]: done

File: .Gruntfile.babel.js (hunk 5 above)
Comment (sandrina-p) [reply .Gruntfile.babel.js:477]: yey thanks @taoeffect!

**Review by mmbotelho — APPROVED** [review]

**Review by taoeffect — APPROVED** [review]
Looks good to me! 👍

**Pull request merged**